app/feature/feature-chat/src/main/kotlin/com/hedvig/android/feature/chat/floating/FloatingBubbleChat.kt

```diff
@@ -0,0 +1,372 @@
+import androidx.compose.ui.layout.LayoutCoordinates
+import androidx.compose.ui.layout.onPlaced
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.Velocity
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.round
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import coil.ImageLoader
+import com.hedvig.android.core.designsystem.material3.DisabledAlpha
+import com.hedvig.android.core.designsystem.material3.squircleLarge
+import com.hedvig.android.core.icons.Hedvig
+import com.hedvig.android.core.icons.hedvig.colored.hedvig.Chat
+import com.hedvig.android.core.icons.hedvig.compose.notificationCircle
+import com.hedvig.android.feature.chat.ChatViewModel
+import com.hedvig.android.feature.chat.ui.ChatDestination
+import com.hedvig.android.navigation.core.HedvigDeepLinkContainer
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
+import org.koin.androidx.compose.koinViewModel
+
+@Composable
+fun FloatingBubbleChat(
+  isInHomeScreen: Boolean,
+  hasUnseenChatMessages: Boolean,
+  imageLoader: ImageLoader,
+  appPackageId: String,
+  hedvigDeepLinkContainer: HedvigDeepLinkContainer,
+  openUrl: (String) -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  val viewModel: ChatViewModel = koinViewModel()
+  val floatingBubbleViewModel: FloatingBubbleViewModel = koinViewModel()
+  val floatingBubbleUiState by floatingBubbleViewModel.uiState.collectAsStateWithLifecycle()
+  val floatingBubbleState = rememberFloatingBubbleState(isInHomeScreen)
+  FloatingBubble(
+    modifier = modifier,
+    floatingBubbleState = floatingBubbleState,
+    showWelcomeTooltip = floatingBubbleUiState.showWelcomeTooltip,
+    onWelcomeTooltipShown = { floatingBubbleViewModel.emit(FloatingBubbleEvent.SeenTooltip) },
+    hasUnseenChatMessages = hasUnseenChatMessages,
+  ) {
+    ChatDestination(
+      viewModel = viewModel,
+      imageLoader = imageLoader,
+      appPackageId = appPackageId,
+      hedvigDeepLinkContainer = hedvigDeepLinkContainer,
+      openUrl = openUrl,
+      onNavigateUp = {
+        floatingBubbleState.minimize()
+      },
+    )
+  }
+}
+
+@Composable
+private fun FloatingBubble(
+  modifier: Modifier = Modifier,
+  floatingBubbleState: FloatingBubbleState,
+  showWelcomeTooltip: Boolean,
+  onWelcomeTooltipShown: () -> Unit,
+  hasUnseenChatMessages: Boolean,
+  expandedContent: @Composable () -> Unit,
+) {
+  val transition = rememberTransition(floatingBubbleState.seekableTransition)
+  floatingBubbleState.PredictiveBackHandler()
+  val sharedChatIcon: @Composable SharedTransitionScope.(
+    AnimatedContentScope,
+    () -> Unit,
+    Modifier,
+  ) -> Unit = { animatedContentScope, onClick, movableContentModifier ->
+    ChatCircle(animatedContentScope, onClick, movableContentModifier)
+  }
+  SharedTransitionLayout(modifier) {
```

Comment:
That's new. So we're okay with using snapshot version for this, bc it's turned off for now anyway?